MantidPlot/src/ProjectRecoveryGUIs/RecoveryFailureView.cpp

```diff
@@ -0,0 +1,131 @@
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2018 ISIS Rutherford Appleton Laboratory UKRI,
+//     NScD Oak Ridge National Laboratory, European Spallation Source
+//     & Institut Laue - Langevin
+// SPDX - License - Identifier: GPL - 3.0 +
+#include "RecoveryFailureView.h"
+#include "ApplicationWindow.h"
+#include "MantidKernel/UsageService.h"
+#include "Script.h"
+#include "ScriptingWindow.h"
+#include "ui_RecoveryFailure.h"
+#include <boost/smart_ptr/make_shared.hpp>
+
+RecoveryFailureView::RecoveryFailureView(QWidget *parent,
+                                         ProjectRecoveryPresenter *presenter)
+    : QDialog(parent), ui(new Ui::RecoveryFailure), m_presenter(presenter) {
+  ui->setupUi(this);
+  ui->tableWidget->horizontalHeader()->setResizeMode(QHeaderView::Stretch);
+  ui->tableWidget->verticalHeader()->setResizeMode(QHeaderView::Stretch);
+  // Make sure the UI has all the data it needs to display
+  m_presenter->fillAllRows();
+  // Set the table information
+  addDataToTable(ui);
+  Mantid::Kernel::UsageService::Instance().registerFeatureUsage(
+      "Interface", "ProjectRecoveryFailureWindow", true);
+}
+
+RecoveryFailureView::~RecoveryFailureView() { delete ui; }
```

Comment:
Why not use `std::unique_ptr`?

Reply:
Same thing as previous when it came to the ui object